verification/extendTCI.go

```diff
@@ -0,0 +1,230 @@
+// Licensed under the Apache-2.0 license
+
+package verification
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"crypto/sha512"
+	"crypto/x509"
+	"hash"
+
+	"testing"
+)
+
+// Check whether the ExtendTCI command updates the current TCI and cumulative TCI.
+func TestExtendTCI(d TestDPEInstance, c DPEClient, t *testing.T) {
+	var err error
+	useSimulation := false // To indicate that simulation context is not used
+
+	// Get default context handle
+	handle := getInitialContextHandle(d, c, t, useSimulation)
+
+	// Get digest size
+	profile, err := GetTransportProfile(d)
+	if err != nil {
+		t.Fatalf("[FATAL]: Could not get profile: %v", err)
+	}
+	digestLen := profile.GetDigestSize()
+
+	// Initialize TCI inputs with all zeroes
+	// since, TCI_DEFAULT for default context is all zeroes
+	defaultTci := make([]byte, digestLen)
+
+	// Initialize hasher
+	var hasher hash.Hash
+	if digestLen == 32 {
+		hasher = sha256.New()
+	} else if digestLen == 48 {
+		hasher = sha512.New384()
+	}
+
+	tciValue := make([]byte, digestLen)
+	for i := range tciValue {
+		tciValue[i] = byte(i)
+	}
+
+	// Set current TCI value
+	_, err = c.ExtendTCI(handle, tciValue)
+	if err != nil {
+		t.Fatalf("[FATAL]: Could not extend TCI: %v", err)
+	}
+
+	// Cross-check current and cumulative measurement by GetTaggedTCI
+	verifyMeasurementsByGetTaggedTCI(c, t, defaultTci, tciValue, hasher)
+
+	// Cross-check current and cumulative measurement by CertifyKey
+	verifyMeasurementsByCertifyKey(c, t, handle, defaultTci, tciValue, hasher)
+}
```

Comment:
Added a new test, instead of using GetTaggedTCI it uses CertifyKey to get TCI_CURRENT/TCI_CUMULATIVE. 
- Get leaf cert from CertifyKey
- Parse MultiTcbInfo 
- Get FWID array and use it to cross check